### PR TITLE
Psr 29 polysync echo command line tool

### PR DESCRIPTION
**Pull request by abstractmachines**
Echo command line utility.

### CODE REVIEW

File: Echo/EchoNode.hpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+     *
+     * PSR-29-F11 : ( PS-153)
+     */
+    std::vector< std::string > getHelpDescriptions();
+
+    /**
+     * @brief Print strings for help page display, append to end of list.
+     * @param std::vector< std:: string > - variable containing help string.
+     */
+    void printHelp( const std::vector< std::string > & helpFlags,
```
Comment (shnewto) [inline Echo/EchoNode.hpp:105]: I feel like the print help utility should exist outside of the node so if we just want to print the help message we don't have to go through the trouble of spinning up the node then cleaning it up immediately after.

File: Echo/EchoNode.cpp [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+    {
+        registerListenerToAllMessageTypes();
+    }
+}
+
+
+void PolySyncEcho::setConfigurationEvent( int argc, char * argv [] )
+{
+    std::string filterFlag = "-filterMessage";
+    std::string echoHeadersOnly = "-echoHeaders";
```
Comment (shnewto) [inline Echo/EchoNode.cpp:42]: Probably there should be a check for invalid command line arguments? So if they enter a flag that doesn't make sense there will be an error that maybe lets them know they can use the help flag? It looks like for now if there is an unexpected flag, like if I misspell "-filterMessage" it just gets ignored without me knowing it didn't take.

File: Echo/EchoNode.cpp [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    else
+    {
+        registerListenerToAllMessageTypes();
+    }
+}
+
+
+void PolySyncEcho::setConfigurationEvent( int argc, char * argv [] )
+{
+    std::string filterFlag = "-filterMessage";
```
Comment (shnewto) [inline Echo/EchoNode.cpp:41]: maybe this should just be -filter. -filterMessage is a little cumbersome and easy for me to mistype.

**Comment by snewt** [comment]
Maybe the filter flag should just be -filter. -filterMessage is a little cumbersome and easy for me to mistype.

**Comment by snewt** [comment]
Probably should be a check for invalid command line arguments? So if they enter a flag that doesn't make sense there will be an error that maybe lets them know they can use the help flag? It looks like for now if there is an unexpected flag, like if I misspell "-filterMessge" it just gets ignored without me knowing it didn't take.

**Comment by snewt** [comment]
I feel like the print help utility should exist outside of the node so if we just want to print the help message we don't have to go through the trouble of spinning up the node then cleaning it up immediately after.

File: Echo/EchoNode.cpp (hunk 2 above)
Comment (abstractmachines) [inline Echo/EchoNode.cpp:42]: true!

**Pull request closed without merging**